markers/stream_excluder.py

```diff
@@ -71,3 +91,25 @@ def pytest_collection_modifyitems(session, config, items):
     # Just to print out the appliance's streams
     if not ssh_connection_failed:
         reporter(config).write("\nAppliance's streams: [{}]\n".format(", ".join(get_streams_id())))
+        # Bail out if the appliance stream or version do not match
+        check_stream = config.getvalue("check_stream")
+        check_version_series = config.getvalue("check_version_series")
+        if check_stream is not None:
+            match = STREAM_RE.match(check_stream)
+            if match is not None:
+                check_stream, major, minor = match.groups()
+                if major and minor:
+                    check_version_series = "{}.{}".join((major, minor))
+            if check_stream.lower() == "upstream":
+                if appliance_is_downstream():
+                    raise Exception("Appliance is downstream when we want upstream one!")
+            elif check_stream.lower() == "downstream":
```

Comment:
And just to clarify, this is a generic question. Any time we compare against a string literal, I will wonder if there's value in having a python type to centralize that comparison behavior.